funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/task_queue_subscriber.py

```diff
@@ -371,28 +385,25 @@ def run(self):
 
         Note: Only one of these options should be used.
         """
+        self.status = SubscriberProcessStatus.running
         try:
             self._connection = self._connect()
             self.event_watcher()
             self._connection.ioloop.start()
         except Exception:
             logger.exception("Failed to start subscriber")
 
-    def close(self) -> None:
-        """This close method needs some rethinking since this whole thing is a separate
-        process with the ioloop callbacks not setting the vars properly
-        """
+    def stop(self, *, block: bool = True) -> None:
+        """stop() is called by the parent to shutdown the subscriber"""
         logger.info("Stopping")
         self.kill_event.set()
-        self._closing = True
-        logger.warning("Waiting for cleanup_complete")
+        if block:
```

Comment:
`stop()` blocks by default, and the tests also only test for that case. Is it worth having the option?

Reply:
I wasn't sure. I added it for that one case thinking I was about to find others to cleanup with it, but then there weren't any. Should I pull it out and make the only behavior blocking?

Reply:
I was just doing some cleanup and removed this. It's easier to remove now and not worry about it -- it can always be added later.